sst/SimengCoreWrapper.cc

```diff
@@ -0,0 +1,213 @@
+// clang-format off
+// DO NOT MOVE FROM TOP OF FILE - https://github.com/sstsimulator/sst-core/issues/865
+#include <sst/core/sst_config.h>
+// clang-format on
+
+#include "SimengCoreWrapper.hh"
+
+#include <cstdlib>
+#include <iostream>
+
+using namespace SST::SSTSimeng;
+using namespace SST::Interfaces;
+
+SimengCoreWrapper::SimengCoreWrapper(SST::ComponentId_t id, SST::Params& params)
+    : SST::Component(id) {
+  output_.init("SimengCoreWrapper[" + getName() + ":@p:@t]:", 999, 0,
+               SST::Output::STDOUT);
+  clock_ = registerClock(params.find<std::string>("clock", "1GHz"),
+                         new SST::Clock::Handler<SimengCoreWrapper>(
+                             this, &SimengCoreWrapper::clockTick));
+
+  // Extract variables from config.py
+  executablePath_ = params.find<std::string>("executable_path", "");
+  executableArgs_ = splitArgs(params.find<std::string>("executable_args", ""));
+  configPath_ = params.find<std::string>("simeng_config_path", "");
+  cacheLineWidth_ = params.find<uint64_t>("cache_line_width", "64");
+  maxAddrMemory_ = params.find<uint64_t>("max_addr_memory", "0");
+
+  if (executablePath_.length() == 0) {
+    output_.fatal(CALL_INFO, 10, 0,
+                  "Simeng executable binary filepath not provided.");
+  }
+  if (maxAddrMemory_ == 0) {
+    output_.fatal(CALL_INFO, 10, 0,
+                  "Maximum address range for memory not provided");
+  }
+
+  iterations_ = 0;
+
+  // Instantiate the StandardMem Interface defined in config.py
+  mem_ = loadUserSubComponent<SST::Interfaces::StandardMem>(
+      "memory", ComponentInfo::SHARE_NONE, clock_,
+      new StandardMem::Handler<SimengCoreWrapper>(
+          this, &SimengCoreWrapper::handleEvent));
+
+  dataMemory_ = std::make_shared<SimengMemInterface>(mem_, cacheLineWidth_,
+                                                     maxAddrMemory_, &output_);
+
+  handlers_ = new SimengMemInterface::SimengMemHandlers(*dataMemory_, &output_);
+
+  // Protected methods from SST::Component used to start simulation
+  registerAsPrimaryComponent();
+  primaryComponentDoNotEndSim();
+}
+
+SimengCoreWrapper::~SimengCoreWrapper() {}
+
+void SimengCoreWrapper::setup() {
+  mem_->setup();
+  output_.verbose(CALL_INFO, 1, 0, "Memory setup complete\n");
+}
+
+void SimengCoreWrapper::handleEvent(StandardMem::Request* ev) {
```

Comment:
What does "ev" stand for?

Reply:
shorthand for ev. I can change it something else?

Reply:
corrected.

Reply:
resolve?